src/main/java/hu/flowacademy/zetaabsencemanager/service/AdminUserService.java

```diff
@@ -37,6 +51,43 @@ public User findOneUser(@NotNull Long id) {
         return userRepository.findByIdAndDeletedAtNull(id).orElseThrow(()->new ResponseStatusException(HttpStatus.BAD_REQUEST, "User not found."));
     }
 
+    public User saveUser(@NotNull User user) {
+        if (StringUtils.isEmpty(user.getFirstName())
+                || StringUtils.isEmpty(user.getLastName())
+                || user.getDateOfBirth() == null
+                || StringUtils.isEmpty(user.getEmail())
+                || user.getDateOfEntry() == null
+                || user.getDateOfEndTrial() == null
+                || user.getIsOnTrial() == null
+                || user.getGroup() == null
+                || StringUtils.isEmpty(user.getPosition())
+                || user.getRole() == null
+                || user.getNumberOfChildren() == null
+                || user.getOtherAbsenceEnt() == null
+        ) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The submitted arguments are invalid.");
+        } else {
+            User newUser = User.builder()
```

Comment:
Miért csinálsz egy új user-t? Amikor a paraméterben kapottat le is menthetnéd?

Reply:
Ehhez nem nyúltam, csak áthelyeztem a függvényeket, hogy azok megfelelő (logikus) sorrendben legyenek (create, update, delete). Egyébként itt a password encodolása miatt csinálunk új usert, de ez változni fog, mert User létrehozásánál nem fogjuk a password-öt kérni. Ez még képlékeny, de Martinék dolgoznak rajta.

Reply:
Ehhez nem nyúltam, csak áthelyeztem a függvényeket, hogy azok megfelelő (logikus) sorrendben legyenek (create, update, delete). Egyébként itt a password encodolása miatt csinálunk új usert, de ez változni fog, mert User létrehozásánál nem fogjuk a password-öt kérni. Ez még képlékeny, de Martinék dolgoznak rajta.